js/components/Star/index.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component, PropTypes } from 'react';
+import {
+  TouchableOpacity,
+  Image,
+  StyleSheet,
+} from 'react-native';
+
+export default class Star extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: false,
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }
+
+  toggle() {
+    this.setState({ value: !this.state.value });
+  }
+
+  render() {
+    const img = this.state.value ?
```

Comment:
Correct dynamic require doesn't work. I think it might be best to import the assets at the top of the file to make this clear.

```js
const images = {
  empty: require('./assets/star-empty.png'),
  full: require('./assets/star-full.png'),
};

... later

render() {
  return <Image source={ this.state.value ? images.full : images.empty }/>
}
```